cmd/topaz/main.go

```diff
@@ -34,17 +45,19 @@ func main() {
 			"container_image":    cc.ContainerImage(),
 			"container_tag":      cc.ContainerTag(),
 			"container_platform": cc.ContainerPlatform(),
-			"container_name":     cc.ContainerName(),
+			"container_name":     cc.ContainerName(ctx.Config.DefaultConfigFile),
 		},
 	)
 
-	ctx, err := cc.NewCommonContext(cli.NoCheck)
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err.Error())
-		os.Exit(1)
-	}
-
 	if err := kongCtx.Run(ctx); err != nil {
 		kongCtx.FatalIfErrorf(err)
 	}
+
+	// only save on config change
+	if app.Contains(kongCtx.Args, "configure") || app.Contains(kongCtx.Args, "-c") || app.Contains(kongCtx.Args, "--config") {
```

Comment:
The `topaz version` command has a `-c` flag too. I don't think we want to save configuration in that case.